osu.Game/Screens/Edit/PromptForSaveDialog.cs

```diff
@@ -17,12 +17,12 @@ public PromptForSaveDialog(Action exit, Action saveAndExit, Action cancel)
 
             Buttons = new PopupDialogButton[]
             {
-                new PopupDialogCancelButton
+                new PopupDialogOkButton
```

Comment:
So... this is why the tests are failing right now, primarily. Previously, the primary button was the "dismiss changes one", now the primary button is the 'save changes" one. And so, the "exit all screens" step at the end of the gameplay test scene, clicks this one now rather than the other one, and ends up saving the beatmap.

Because it ends up saving the beatmap, and `IsolateSavingFromDatabase` is set to `false` in the gameplay test scene, that has now polluted the common database/file store, and the next test in sequence that is expecting to see an unmodified beatmap fails.

Moreover, you've reported [seeing the same on `master`](https://discord.com/channels/188630481301012481/188630652340404224/955484661965000704), but that is actually backwards - because non-isolated test scenes use *the client database* when ran in visual test browser, the pollution actually *backported* to master. To add insult to injury, tear down tries to remove the beatmap... but that just ends up doing a soft-delete, and a reimport "wisely" determines that the soft-deleted beatmap can be reused, bringing the broken beatmap back from the dead.

I was aware that this "isolate saving from beatmap" flag was dicey at best, but this is a failure of rather epic proportions. Right now I can offer [this quick fix branch](https://github.com/bdach/osu/tree/gameplay-test-beatmap-pollution) that just turns off the `IsolateSavingFromDatabase` flag on the affected test scene and locally simulates a save. But that is probably insufficient and this needs to be addressed properly...

Reply:
Can confirm that this is the case, which for now is enough to let me work around it. The actual issue of save being the default on-exit behaviour remains the problem (which I suspected, but was very confused by the nature of these test failures).

Which brings up a follow up question: in most apps where confirm dialogs are presented to prevent data loss, hitting the exit key twice will close the dialog but not perform an action. I think we likely want the same here, right? May require some further thinking with the exit process of editor screens in this case.

Reply:
I'll keep the existing behaviour on this PR to reduce complexity, but would appreciate feedback on this to make sure everyone agrees with the above. I'll make that change separately if so.